### DISCUSSION TITLE
0.10.0 Beta 5 Performance Regression

### DISCUSSION THREAD

**Discussion by Firerouge** [blakeblackshear/frigate#2436] [General]
There's been a noticeable performance regression between beta 4 and 5
![Screen Shot 2021-12-12 at 8 39 57 PM](https://user-images.githubusercontent.com/9072079/145753539-1396cdce-94f1-4d8e-a19b-cbaa0e71b7bb.png)

Slightly more specifically the process `python3 -u -m frigate` now consistently uses 90-100% of one core.
![Screen Shot 2021-12-12 at 8 43 59 PM](https://user-images.githubusercontent.com/9072079/145753858-b959c156-0c5b-4925-b12d-626aa93c9590.png)
![Screen Shot 2021-12-12 at 8 46 16 PM](https://user-images.githubusercontent.com/9072079/145753944-ba93035a-776a-4e7e-835b-353477da5894.png)

**Comment by blakeblackshear** [comment]
It does run a check to ensure the recordings in the database are still on disk during startup. See if it goes back down after a while.

**Reply by Firerouge** [reply]
That's probably it 92e08b92f580b03cf1f0eb6bdbe60b3e397dbcf2 CPU usage recedes to a similar level after about seven minutes. 
Can the check be disabled, deferred to a later time, or ran when idle of activity?

Perhaps the new sync process should be in logger.info to make this all a bit clearer.
I think [this section which deletes events](https://github.com/blakeblackshear/frigate/blob/92e08b92f580b03cf1f0eb6bdbe60b3e397dbcf2/frigate/record.py#L438) from the database should at least be logged info or warn, couldn't missing recording files be an indicator of further errors?

**Reply by blakeblackshear** [reply]
I am just going to disable it for now and push a beta 6. I will reevaluate running it differently later.

**Comment by ozett** [comment]
is this here discussion of beta5?
where do i find the temperatur-stats for the corals?

changelog: "Report coral temperature stats when available"
on the gui?

edit: found it in mqtt: https://github.com/blakeblackshear/frigate/commit/a5c13e7455e37cce9a2df2e1b3b09420ad3bfb5c
thanks

edit2: temps were gone, after a kernel update of my ubuntu. i had to re-install the driver: 
```apt-get --reinstall install gasket-dkms libedgetpu1-std```

**Reply by ozett** [reply]
you could also determine the frequency from the temp.
after that information here: https://github.com/google-coral/edgetpu/issues/197

**Comment by Minglarn** [comment]
Wow! This new retain method for recording is absolutely  a brilliant solution! 
Instead of save recordings 24/7, it saves recordings where there was motion detected. 
Good job Blake! This is a brilliant solution!
